raiko-block-downloader/src/preflight/util.rs

```diff
@@ -0,0 +1,597 @@
+use alloy_primitives::{hex, Log as LogStruct, B256};
+use alloy_provider::{Provider, ReqwestProvider};
+use alloy_rpc_types::{Filter, Header, Log, Transaction as AlloyRpcTransaction};
+use alloy_sol_types::{SolCall, SolEvent};
+use anyhow::{anyhow, bail, ensure, Context, Result};
+use kzg::kzg_types::ZFr;
+use kzg_traits::{
+    eip_4844::{blob_to_kzg_commitment_rust, Blob},
+    Fr, G1,
+};
+use raiko_lib::{
+    builder::{OptimisticDatabase, RethBlockBuilder},
+    clear_line,
+    consts::ChainSpec,
+    inplace_print,
+    input::{
+        ontake::{BlockProposedV2, CalldataTxList},
+        proposeBlockCall, BlobProofType, BlockProposed, BlockProposedFork, TaikoGuestInput,
+        TaikoProverData,
+    },
+    primitives::eip4844::{self, commitment_to_version_hash, KZG_SETTINGS},
+};
+use reth_evm_ethereum::taiko::{decode_anchor, decode_anchor_ontake};
+use reth_primitives::Block;
+use reth_revm::primitives::SpecId;
+use serde::{Deserialize, Serialize};
+use tracing::{debug, error, info, warn};
+
+use crate::{provider::{db::ProviderDb, rpc::RpcBlockDataProvider, BlockDataProvider}, RaikoResult};
+
+/// Optimize data gathering by executing the transactions multiple times so data can be requested in batches
```

Comment:
Could you please extend the comment to elaborate why "executing the transactions multiple times" optimizes data gathering?